File: project/security/src/main/java/org/tby/fourdk/security/RoleReadRepository.java
package org.tby.fourdk.security;

import java.util.List;
import java.util.Optional;

public interface RoleReadRepository {

    List<Role> findAllRoles();

    AppPermissions findAllPermissions();

    Optional<Role> findRole(Role role);

    List<Permission> findPermissionsForGivenRoles(Role role);

}

File: rust/project/security-rust/src/dddk/security/authorized_strategy_impl/role_read_repository.rs
use crate::dddk::security::permission::Permission;
use crate::dddk::security::role::Role;

pub trait RoleReadRepository {
    fn find_all_roles(&self) -> Vec<Role>;

    fn find_all_permissions(&self) -> Vec<Permission>;

    fn find_role_by_name(&self, name: String) -> Option<Role>;

    fn find_permissions_for_given_role(&self, name: String) -> Vec<Permission>;
}